src/fides/api/ops/api/deps.py

```diff
@@ -30,7 +30,11 @@ def get_api_session() -> Session:
     """Gets the shared database session to use for API functionality"""
     global _engine  # pylint: disable=W0603
     if not _engine:
-        _engine = get_db_engine(config=CONFIG)
+        _engine = get_db_engine(
+            config=CONFIG,
+            pool_size=CONFIG.database.api_engine_pool_size,
+            max_overflow=CONFIG.database.api_engine_max_overflow,
```

Comment:
I think this will work well for the API endpoints ops-side but the ctl side creates their own synchronous and asynchronous engines, that aren't taking in these settings - alternatively this might be a good time to unify if we're trying to decrease the number of engines created overall:

The ctl side has a separate [get_db_engine](https://github.com/ethyca/fides/blob/main/src/fides/core/utils.py#L65) method that a few of their API endpoints call, and a lot of their other endpoints use the [async or sync engines created here](https://github.com/ethyca/fides/blob/main/src/fides/api/ctl/database/session.py)

I stumbled on this in my updates  to populate the application db instead of the test db when running the application locally https://github.com/ethyca/fides/pull/1731, but didn't go all the way through with trying to unify this between ctl/ops.